### ISSUE TITLE
Add aria-labels to mobile menu buttons

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#617]
- main menu
- add menu
- review all dropdowns in navbar
Related to #526

**Comment by joshdanielson** [comment]
@IagoSRL I think we should use aria-labelledby instead (it seems):
https://www.w3.org/TR/WCAG20-TECHS/aria#ARIA13

I think I understand what you're talking about now with the smartNav. So the main menu and add menu would become components that would be rendered in each activity vs in the appNav.html?

**Comment by joshdanielson** [comment]
Here's some more info: https://www.w3.org/TR/wai-aria-practices/examples/landmarks/navigation.html

**Comment by IagoSRL** [comment]
> I think I understand what you're talking about now with the smartNav. So the main menu and add menu would become components that would be rendered in each activity vs in the appNav.html?

Yes, and the full smartNav actually.

Using components and child components, "may" end as something like
```html
<!-- Typical at main activities for logged users -->
<smart-nav-bar>
    <main-menu slot="left"></main-menu>
    <h1 slot="header">Activity name</h1>
    <add-menu slot="right"></add-menu>
</smart-nav-bar>

<!-- Another examples -->
<smart-nav-bar>
    <smart-nav-bar-back-button slot="left"></smart-nav-bar-back-button>
    <smart-nav-bar-logo slot="header"></smart-nav-bar-logo>
    <smart-nav-bar-button slot="right" params="action: customActivityAction"></smart-nav-bar-button>
</smart-nav-bar>

<smart-nav-bar>
    <a href="#!home" slot="left">Back to home</a>
    <h1 slot="header">Hello</h1>
    <button slot="right" type="button" data-bind="click: doSomething">Something</button>
</smart-nav-bar>
```
*the examples are an initial idea/proposal. The slot attribute let the container component to know where belongs each children*

There would be some dynamics things too, for example displaying a smart-nav-bar if user logged and another if anonymous (at public pages), and even that logic may be wrapped around another component to make it easier for common cases; but the good, is that every activity has its own nav-bar instance.

**Comment by IagoSRL** [comment]
About *aria-labelledby*, from my understanding, that must be used if the label already exists at another (near, related) element in the page; then screen readers may avoid read twice the label (once in the element and another one in the *aria-label*).

**Comment by joshdanielson** [comment]
@IagoSRL yes, I think that sounds right. so each activity would have one nav element with nested ul/li elements and an h tag. check out appNav.html in the is528 branch. I think it's clean and ready for the components. Just need to make sure the proper tags are included and that we can have the dynamic aria-labelledby and corresponding h tags (with sr-only class). We won't use title or aria-label so it's not redundant.

**Comment by joshdanielson** [comment]
@IagoSRL new proposal based on the courses I've been taking. We can discuss. 
- all h1 tags go to <main> section and removed from navbars (like on desktop) and replaced with the link/logo to the home page (I can edit the title in the javascript to null). They say there are exceptions, but I don't think this qualifies and it much clearer to the user. 
- the mobile menu must be a button (it is now) and have an aria-label="Main menu". 
- any back links should be <a> elements (I'm not sure they are now)
- I changed my mind about the aria-labelledby on the navigation. The aria-label is within guidelines and cleaner.

So I'm not actually sure it needs a lot of refactoring.
